pkg/terraform/gather/ovirt/ip.go

```diff
@@ -2,13 +2,105 @@
 package ovirt
 
 import (
+	//"context"
+	"fmt"
 	"github.com/openshift/installer/pkg/terraform"
+	"github.com/openshift/installer/pkg/asset/installconfig/ovirt"
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"github.com/sirupsen/logrus"
+	"net"
+	//"time"
 )
 
+
+func findVirtualMachineIP(c *ovirtsdk4.Connection,moRefValue string,subnetCIDR string) (string, error) {
+
+	vmsService := c.SystemService().VmsService()
+	// Get the reference to the vm service for a particular vm:
+	// vmsService := conn.SystemService().VmsService()
+
+	_, subnet, _ := net.ParseCIDR(subnetCIDR)
+
+	//# Look up fot the vm by id:
+	vmResp, err := vmsService.List().Search(fmt.Sprintf("id=%s",moRefValue)).Send()
+	if err != nil {
+		fmt.Printf("Failed to search vm list, reason: %v\n", err)
+		return "",fmt.Errorf("aa")
+	}
+	vmSlice, _ := vmResp.Vms()
+	vm := vmSlice.Slice()[0]
+	ipAddress := ""
+	// Get the reported-devices service for this vm:
+	reportedDevicesService := vmsService.VmService(vm.MustId()).ReportedDevicesService()
+
+	// Get the guest reported devices
+	reportedDeviceResp, err := reportedDevicesService.List().Send()
+	if err != nil {
+		fmt.Printf("Failed to get reported devices list, reason: %v\n", err)
+		return "",fmt.Errorf("aa")
+	}
+	reportedDeviceSlice, _ := reportedDeviceResp.ReportedDevice()
+
+	if len(reportedDeviceSlice.Slice())==0{
+		return "",fmt.Errorf("IPS not found")
+	}
+
+	for _, reportedDevice := range reportedDeviceSlice.Slice() {
+		ips := reportedDevice.MustIps()
+
+		if ips != nil {
+			for _, ip := range ips.Slice() {
+				ipres := net.ParseIP(ip.MustAddress())
+				if ipres != nil {
+					if subnet.Contains(ipres){
+						ipAddress = ip.MustAddress()
+					}
+				}
+
+			}
+		}
+	}
+
+	logrus.Debugln(fmt.Sprintf("ovirt vm id: %s , found bootstrap IP %s",moRefValue,ipAddress))
+	return ipAddress, nil
+}
+
+
 // BootstrapIP returns the ip address for bootstrap host.
 // still unsupported, because qemu-ga is not available - see https://bugzilla.redhat.com/show_bug.cgi?id=1764804
 func BootstrapIP(tfs *terraform.State) (string, error) {
-	return "", nil
+
+	client, err := ovirt.NewConnection()
+	if err != nil {
+		return "",fmt.Errorf("failed to initialize connection to ovirt-engine's %s", err)
+	}
+	defer client.Close()
+
+	br, err := terraform.LookupResource(tfs, "module.bootstrap", "ovirt_vm", "bootstrap")
+
+	if err != nil {
+		return "", errors.Wrap(err, "failed to lookup bootstrap")
+	}
+
+	if len(br.Instances) == 0 {
+		return "", errors.New("no bootstrap instance found")
+	}
+
+	vmid, found, err := unstructured.NestedString(br.Instances[0].Attributes, "id")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to lookup bootstrap managed object reference")
+	}
+	if !found {
+		return "", errors.Errorf("failed to lookup bootstrap managed object reference")
+	}
+
+	ip, err := findVirtualMachineIP(client, vmid,"192.168.0.0/16")
```

Comment:
this line is temp for testing , we need to decide about it ..

Reply:
we can examine the api_vip from the config and prefer the choose the bootstrap address in the same network.

Reply:
this is very similar to what is done for choosing the node-ip: https://github.com/openshift/baremetal-runtimecfg/blob/5ba9497f109f6d3170ecc7f0f879e93090cf8eda/pkg/utils/addresses.go#L160-L165